### PR TITLE
[stdlib] Adjust _assertionFailed signature for naming guidelines

### PR DESCRIPTION
**Pull request by airspeedswift**
Just internal changes, no externally-facing API changes. Fixes ABI FIXMEs #18, 19 and 20

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/AssertCommon.swift [hunk 1]
```diff
@@ -112,10 +111,10 @@ func _assertionFailed(
 @_versioned
 @inline(never)
 @_semantics("stdlib_binary_only")
-func _assertionFailed(
+func _assertionFailure(
   // FIXME(ABI)#19 : add argument labels to conform to API guidelines.
```
Comment (slavapestov) [inline stdlib/public/core/AssertCommon.swift]: This fixme should be removed too right?

Comment (airspeedswift) [inline stdlib/public/core/AssertCommon.swift]: Ah, yes, thanks!

**Comment by gottesmm** [comment]
You should double check that there are no optimizer dependencies on this name. I think we got rid of them all, but it wouldn't hurt to check.

**Comment by airspeedswift** [comment]
@gottesmm I did a grep for `_assertionFail` across the swift repo when I did it and these are the only occurrences, is that enough or might there be some hidden gotcha?

**Comment by gottesmm** [comment]
The mangled name is what you would need to look for.

**Comment by jrose-apple** [comment]
The regular name will always be in the mangled name in the old mangling, so I think we're okay. (And going forward, we should not pattern-match mangled names in the optimizer. We can use `@_semantics` for that.)

**Comment by shahmishal** [comment]
Testing PR job change: (Please ignore this trigger)
@swift-ci Please smoke test

**Pull request merged**